### PR TITLE
[BH-1932] Add What's New application

### PR DESCRIPTION
**Pull request by Lefucjusz**
Added What's New application displaying
most important changes between versions
after updating to the new one.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

**Review by dawidwojtas-mudita — APPROVED** [review]
LGTM
Remove unused code before merging ;)

File: products/BellHybrid/services/db/agents/WhatsNewAgent.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            ret.push_back({.description{(*retQuery)[0].getString()}, .iconName{(*retQuery)[1].getString()}});
+        } while (retQuery->nextRow());
+
+        return ret;
+    }
+} // namespace
+
+namespace service::db::agents
+{
+    WhatsNew::WhatsNew(sys::Service *parentService, const std::string dbName)
```
[the remark this answers is not in the document]
Comment (dawidwojtas-mudita) [reply products/BellHybrid/services/db/agents/WhatsNewAgent.cpp]: It's outdated already

File: products/BellHybrid/services/db/ServiceDB.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ namespace
     constexpr auto multimediaDatabaseName{"multimedia.db"};
     constexpr auto settingsDatabaseName{"settings_bell.db"};
     constexpr auto meditationStatsDatabaseName{"meditation_stats.db"};
-    constexpr auto whatsNewDatabaseName{"whats_new.db"};
+    constexpr auto whatsNewDatabaseName{"whats-new.db"};
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/services/db/ServiceDB.cpp]: Fixed

File: products/BellHybrid/apps/application-bell-whats-new/presenter/WhatsNewMainPresenter.hpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017-2024, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#pragma once
+
+#include <apps-common/BasePresenter.hpp>
+#include <string>
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/apps/application-bell-whats-new/presenter/WhatsNewMainPresenter.hpp]: You're right, removed

File: products/BellHybrid/apps/common/src/options/BellOptionWithDescriptionWindow.cpp [hunk 4]
```diff
@@ -80,19 +79,27 @@ namespace gui
         body->resizeItems();
     }
 
-    void BellOptionWithDescriptionWindow::setListDescription(const std::string &title)
+    void BellOptionWithDescriptionWindow::setListDescription(const std::string &title,
+                                                             TokenMap tokenMap,
+                                                             const std::string &font,
+                                                             const unsigned height)
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/apps/common/src/options/BellOptionWithDescriptionWindow.cpp]: I must have had different idea for this argument initially (something passed by ref probably) and just forgot to remove this const, fixed :wink:

File: products/BellHybrid/apps/application-bell-whats-new/models/WhatsNewModel.cpp [hunk 5]
```diff
@@ -2,10 +2,63 @@
...
+    WhatsNewModel::WhatsNewModel(app::ApplicationCommon *app) : app{app}
+    {
+        const auto version = getVersionNumber(VERSION);
+        if (!version.has_value()) {
+            return;
+        }
+        const auto result = sendDBRequest(app, std::make_shared<messages::GetByVersion>(version.value()));
+        if (result.has_value()) {
+            for (auto &record : result->records) {
+                LOG_ERROR("*** changes: %s iconName: %s ***", record.description.c_str(), record.iconName.c_str());
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/apps/application-bell-whats-new/models/WhatsNewModel.cpp]: It's already removed, this file doesn't exist finally

File: products/BellHybrid/services/db/agents/WhatsNewAgent.cpp [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+    sys::MessagePointer WhatsNew::handleGetRecordsByVersion(const sys::Message *req)
+    {
+        if (auto msg = dynamic_cast<const whatsNew::messages::GetByVersion *>(req)) {
+            const auto records = getRecordsByVersion(&db, msg->version);
+            return std::make_shared<whatsNew::messages::Response>(records);
+        }
+        return std::make_shared<sys::ResponseMessage>();
+    }
+} // namespace service::db::agents
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/services/db/agents/WhatsNewAgent.cpp]: Done

File: products/BellHybrid/services/db/include/db/WhatsNewMessages.hpp [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            Response() = default;
+            explicit Response(const std::vector<Record> &records) : records{records}
+            {}
+
+            std::vector<Record> records{};
+        };
+
+    } // namespace messages
+} // namespace service::db::whatsNew
```
[the remark this answers is not in the document]
Comment (Lefucjusz) [reply products/BellHybrid/services/db/include/db/WhatsNewMessages.hpp]: Done

**Review by Maciej-Mudita — APPROVED** [review]

**Pull request merged**